### DISCUSSION TITLE
Different footsteps per PhysMat?

### DISCUSSION THREAD

**Discussion by xkamronx** [Sixze/ALS-Refactored#268] [Q&A] [answered]
I had it working with the standard community edition, but Refactored seems to have changed how it functions; now it uses the specific DataAsset listed in the Notify without the ability to expand it easily to play different footsteps based on the physical material from the trace.

Any tips / help with getting this set up?

Edit - Super easy to do! Thank you @woodencase01 !

**Comment by woodencase01** [comment, marked as answer]
Look for the "AlsFoorstepEffectsSettings", there is an array where you can change the particle system, decal, and sound. You need to make one metasound per material. It's actually very well made!
![image](https://user-images.githubusercontent.com/4787041/228208939-94b82114-6d92-4eb7-8834-d9b6195a0ee6.png)

**Comment by xkamronx** [comment]
Thank you so much! That is super easy once I understand how to do it, heck yeah.

**Comment by Magic-Summer** [comment]
In case anyone got lost like me, this uses the `struct` in
`Plugins/ALS-Refactored/Source/ALS/Public/Notifies/AlsAnimNotify_FootstepEffects.h`
The Data Assets itself is in `Plugins/ALS-Refactored/Content/ALS/Data/Footstep/FES_Als_Default.uasset`
To add custom footsteps without changing the plugin code one needs to add a [core-redirect](https://github.com/Sixze/ALS-Refactored/discussions/395)
